lib/trino-parquet/src/main/java/io/trino/parquet/reader/PageReader.java

```diff
@@ -37,23 +40,27 @@
 import static com.google.common.base.Preconditions.checkState;
 import static io.trino.parquet.ParquetCompressionUtils.decompress;
 import static io.trino.parquet.ParquetReaderUtils.isOnlyDictionaryEncodingPages;
+import static org.apache.parquet.format.CompressionCodec.UNCOMPRESSED;
 
 public final class PageReader
 {
     private final CompressionCodec codec;
     private final boolean hasOnlyDictionaryEncodedPages;
     private final boolean hasNoNulls;
     private final PeekingIterator<Page> compressedPages;
+    private final LocalMemoryContext decompressionBufferMemoryUsage;
 
     private boolean dictionaryAlreadyRead;
     private int dataPageReadCount;
+    private byte[] decompressorOutputBuffer;
```

Comment:
Can reusing of buffer interfere with block laziness (e.g. produced `Blocks` become stalled)?